MimeTypeMap.cs

```diff
@@ -761,6 +762,11 @@ public static bool TryGetMimeType(string str, out string mimeType)
                 str = Dot + str;
             }
 
+            if (str.Contains(QuestionMark))

```

Comment:
Your approach causes the complete string to be searched multiple times. You could prevent this by rewriting the check as follows.

```cs
var indexQuestionMark = str.IndexOf(QuestionMark, StringComparison.Ordinal);
if (indexQuestionMark != -1) {
    str = str.Remove(indexQuestionMark);
}
```